skore/src/skore/project/_launch.py

```diff
@@ -0,0 +1,358 @@
+
+    @classmethod
+    def rejoin(cls, project: Project):
+        """Rejoin a project to a server.
+
+        Parameters
+        ----------
+        project : Project
+            The project to associate with the server.
+
+        Returns
+        -------
+        ServerInfo
+            The server information.
+        """
+        pid_file = cls._get_pid_file_path(project)
+        if not pid_file.exists():
+            return
+
+        info = json.load(pid_file.open())
+        return cls(project, info["port"], info["pid"])
+
+    def save_pid_file(self):
+        """Save server information to disk."""
+        info = {"port": self.port, "pid": self.pid}
+        self.pid_file.parent.mkdir(parents=True, exist_ok=True)
+        with open(self.pid_file, "w") as f:
+            json.dump(info, f)
+
+    def delete_pid_file(self):
+        """Delete server information from disk."""
+        with contextlib.suppress(FileNotFoundError):
+            os.remove(self.pid_file)
+
+    def load_pid_file(self) -> Union[dict, None]:
+        """Load server information from disk if it exists."""
+        try:
+            with open(self.pid_file) as f:
+                return json.load(f)
+        except (FileNotFoundError, json.JSONDecodeError):
+            return None
+
+
+def is_server_started(port: int, timeout: float = 10.0, interval: float = 0.1) -> bool:
+    """Wait for server to be ready by attempting to connect to the port.
+
+    Parameters
+    ----------
+    port : int
+        The port to check.
+    timeout : float, default=10.0
+        Maximum time to wait in seconds.
+    interval : float, default=0.1
+        Time between connection attempts in seconds.
+
+    Returns
+    -------
+    bool
+        True if server is ready, False if timeout occurred.
+    """
+    start_time = time.time()
+    while time.time() - start_time < timeout:
+        try:
+            with socket.create_connection(("localhost", port), timeout=interval):
+                return True
+        except (socket.timeout, ConnectionRefusedError):
+            time.sleep(interval)
+    return False
+
+
+def start_server_in_subprocess(project: Project, port: int, open_browser: bool):
+    """Start the server in a separate process.
+
+    Parameters
+    ----------
+    project : Project
+        The project to associate with the server.
+    port : int
+        The port to use for the server.
+    open_browser : bool
+        Whether to open the browser.
+
+    Returns
+    -------
+    subprocess.Popen
+        The process running the server.
+
+    Raises
+    ------
+    RuntimeError
+        If the server failed to start within the timeout period.
+    """
+    cmd = [
+        sys.executable,
+        "-m",
+        "skore.ui.server",
+        "--port",
+        str(port),
+        "--project-path",
+        str(project.path),
+    ]
+
+    process = subprocess.Popen(
+        cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True
```

Comment:
Don't you need to use `/dev/null` for `stdin`, `stdout` and `stderr` to detach the server process from the parent process and make it survive after jupyter kernel restart or script ending (daemon-like process)?

Reply:
I manually handle it:

```python
        if keep_alive:
            atexit.register(block_before_cleanup, project, process)
        else:
            atexit.register(cleanup_server, project)
```

In this case, I can rejoin the server process and request a KeyboardInterrupt to cleanup. I find it explicit.

>  make it survive after jupyter kernel restart or script ending

I would expect this behaviour for the script but it was not my first thought for jupyter: I would expect that when restarting the kernel, I'll go and rerun a set of cell and I'll re-execute the `skore.open` that is going to restart the server.

But I'm open to changes on this side.